model/Offer.ts

```diff
@@ -0,0 +1,22 @@
+export type Offer = {
```

Comment:
```suggestion
export type OfferInfo = {
```
Maybe OfferInfo is a better name, in this case you are decode the offer!